docs/source/indexing.rst

```diff
@@ -0,0 +1,82 @@
+Indexing Rules
+==============
+
+Data objects can be indexed similarly to NumPy arrays or tensors.
+
+1. **Indexing as tuples**
+
+   - As in NumPy, an index is always a tuple of indices. If it is not, it is converted to a tuple. The elements of the tuple are matched to the dimensions of the data from left to right.
+
+   - Example::
+
+       data[0, 1] == data[(0, 1)]  # The index is converted to a tuple
+       data[(0, 1), ] != data[(0, 1)] 
+	   data[(0, 1), ] == data[((0, 1), )]
+
+2. **Broadcasting**
+
+   - Indexing operations are performed on the fields of a dataclass in the following sequence:
+
+     1. Broadcasting to the dataclass's shape.
+     2. Indexing.
+     3. Replacing stride-0 dimensions with singletons wherever possible.
+
+   - This means each data object can be indexed as if all fields had the shape returned by ``dataobject.shape``, even if the fields are only broadcastable to this shape.
+
+   - Example: ``acqidx`` is singleton in ``k0``. But the parent kdata object can be indexed along k0.
+
+3. **Consistency of shapes after indexing**
+
+   - Two data objects with the same shape property, indexed by the same indexer, will result in the same shape property.
+   - This does *not* mean that all fields of the data class have the same shape after indexing, but rather that the fields can be broadcasted to the common shape.
+
+4. **Slice indexing**
+
+   - Indexing with a slice object or colon syntax (e.g., ``0:3``) selects a slice of the data.
+   - A view along the dimension is returned if the stride is positive.
+
+   - Example::
+
+       a[slice1, slice2] == a[slice1, :][:, slice2] == a[:, slice1][slice1, :]
+
+   - For negative strides, a copy is performed because the underlying ``torch.Tensor`` does not support negative strides. This differs from NumPy, which always returns a view, and PyTorch, which raises an error for negative strides.

```

Comment:
i was considering that, but would you expect an error or rely on the error?
this is strictly a superset of pytorch.

Reply:
my only concern would be that people more used to numpy often use negative strides and would expect a view and it would not be obvious to them that we do a copy. If the user knew that a copy is made then they might use a different approach which does not require a copy. Maybe we could print a warning?

Reply:
Warning is a good idea!

Reply:
..i will skip this for now and just raise an exception